security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -213,6 +268,41 @@ func testHelper(t *testing.T, arg ca2.Options, cb secretCallback, testInvalidRes
 	checkStaledConnCount(t)
 }
 
+func testCredentialFetcherHelper(t *testing.T, arg ca2.Options, cb secretCallback, testEmptyToken bool) {
```

Comment:
Instead of using testEmptyToken, just pass the token to testCredentialFetcherHelper(), and set the token into mockSecretStore{ expectedToken: token }. Let mockSecretStore.GenerateSecret() compare passed token with expectedToken